python/pyarrow/tests/test_ipc.py

```diff
@@ -416,6 +416,15 @@ def test_pandas_serialize_round_trip_not_string_columns():
     assert_frame_equal(result, df)
 
 
+def test_serialize_pandas_no_preserve_index():
+    df = pd.DataFrame({'a': [1, 2, 3]}, index=[1, 2, 3])
+    expected = pd.DataFrame({'a': [1, 2, 3]})
+
+    buf = pa.serialize_pandas(df, preserve_index=False)
+    result = pa.deserialize_pandas(buf)
```

Comment:
I wonder if `deserialize_pandas` should also have this flag, though I suppose a `reset_index()` call could always be made.

Reply:
Conceivably, if this ends up being useful we can add it later, but we'd need to twiddle the parameter down a bit deeper than this function